ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/helpers/MiscHelpers.java

```diff
@@ -114,15 +114,23 @@ public int computeProposerIndex(
     }
   }
 
-  public UInt64 computeEpochAtSlot(UInt64 slot) {
+  public UInt64 computeEpochAtSlot(final UInt64 slot) {
     return slot.dividedBy(specConfig.getSlotsPerEpoch());
   }
 
-  public UInt64 computeStartSlotAtEpoch(UInt64 epoch) {
+  public UInt64 computeStartSlotAtEpoch(final UInt64 epoch) {
     return epoch.times(specConfig.getSlotsPerEpoch());
   }
 
-  public UInt64 computeTimeAtSlot(BeaconState state, UInt64 slot) {
+  public UInt64 computeSlotAtTime(final UInt64 genesisTime, final UInt64 currentTime) {
+    return currentTime.minusMinZero(genesisTime).dividedBy(specConfig.getSecondsPerSlot());
+  }
+
+  public UInt64 computeTimeAtSlot(final UInt64 genesisTime, final UInt64 slot) {
+    return genesisTime.plus(slot.times(specConfig.getSecondsPerSlot()));
+  }
+
+  public UInt64 computeTimeAtSlot(final BeaconState state, final UInt64 slot) {
```

Comment:
This interface i really don't like and its not used much, maybe we just change the couple of occurrences to pass in genesis time...

If we were passing in state i'd expect that we just pass state (which has slot)

Since we're passing in state just to get the genesis time, it's just confusing things, and it'd be better to just not have it and let ppl call interface at 129.

Reply:
Agree. Will change.